cmd/thanos/store.go

```diff
@@ -84,6 +84,9 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application) {
 	consistencyDelay := modelDuration(cmd.Flag("consistency-delay", "Minimum age of all blocks before they are being read. Set it to safe value (e.g 30m) if your object storage is eventually consistent. GCS and S3 are (roughly) strongly consistent.").
 		Default("0s"))
 
+	deleteDelay := modelDuration(cmd.Flag("delete-delay", "Time before a block marked for deletion is deleted from bucket").
```

Comment:
In this case store is not really deleting anything so users can be confused. I would make this `ignore-deletion-marks-delay` and explain fully why we do it and that value has to be half of delete delay for compactor. (:

Reply:
We need to create `ignoreDeletionMarkFilter` with deleteDelay, so we are using the deleteDelay input from the user. 

Yes, we can rename this to `ignore-deletion-marks-delay`, but then there are multiple configurations for people to remember and that is the reason why I didn't want to add an extra config.

Reply:
This is really trying to connect two things together. I would say `ignore-deletion-marks-delay` is important as store does not care about deletion, just about when to ignore those. That's name closer to functionality makes more sense.

Reply:
I have made this change, There can be some feedback point on explanation about `ignore-deletion-marks-delay`, so I am keeping this conversation open.

Reply:
The feedback is provided on different thread here: https://github.com/thanos-io/thanos/pull/2136#discussion_r387918072
Closing this conversation, please create a new one if there are any points.